runtime/lua/vim/filetype/detect.lua

```diff
@@ -1055,6 +1086,18 @@ function M.web(bufnr)
   return 'winbatch'
 end
 
+-- XFree86 config
+function M.xfree86(bufnr)
+  local line = getlines(bufnr, 1)
+  local on_detect
+  if matchregex(line, [[\<XConfigurator\>]]) then
+    on_detect = function(b)
+      vim.b[b].xf86conf_xfree86_version = 3
+    end
+  end
+  return 'xf86conf', on_detect
```

Comment:
Might as well stuff the `matchregex` and `getlines` call into the `on_detect` function, as it doesn't affect the returned filetype at all.

```suggestion
  return 'xf86conf', function(b)
    local line = getlines(b, 1)
    if matchregex(line, [[\<XConfigurator\>]]) then
      vim.b[b].xf86conf_xfree86_version = 3
    end
  end
```

Reply:
(could even do without the `local line`; just inline the `getlines(b ,1)`)